From 1a7087a110215cd85e3f6969761d7b564fc3cb97 Mon Sep 17 00:00:00 2001
From: Joshua Watt <JPEWhacker@gmail.com>
Date: Tue, 30 May 2023 10:12:11 -0500
Subject: [PATCH 09/22] classes/create-spdx-2.2: Respect PKG for providers

If a package renames itself using PKG, the new name should be respected
as a name that the recipe provides.

(From OE-Core rev: 871f48126180bf30c001c6b97268cee1853dbda2)

Signed-off-by: Joshua Watt <JPEWhacker@gmail.com>
Signed-off-by: Richard Purdie <richard.purdie@linuxfoundation.org>
---
 meta/classes/create-spdx-2.2.bbclass | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/meta/classes/create-spdx-2.2.bbclass b/meta/classes/create-spdx-2.2.bbclass
index 6ec0c1465e..f12e155f6b 100644
--- a/meta/classes/create-spdx-2.2.bbclass
+++ b/meta/classes/create-spdx-2.2.bbclass
@@ -688,6 +688,10 @@ def collect_package_providers(d):
             rprovides = set(n for n, _ in bb.utils.explode_dep_versions2(pkg_data.get("RPROVIDES", "")).items())
             rprovides.add(pkg)
 
+            if "PKG" in pkg_data:
+                pkg = pkg_data["PKG"]
+                rprovides.add(pkg)
+
             for r in rprovides:
                 providers[r] = (pkg, dep_hashfn)
 
